src/uu/tail/src/args.rs

```diff
@@ -362,6 +362,72 @@ fn parse_num(src: &str) -> Result<Signum, ParseSizeError> {
     })
 }
 
+/// Parses a string into a [`Duration`] by accepting a [`f64::MAX`] value in the `src` string.
+/// However, internally the parsed [`Duration`] is capped at `seconds == u64::MAX`, `nanos (max) ==
+/// .999999999` and `nanos (min if not 0) == .000000001`. Infinity values like `inf`, `+infinity`
+/// etc. are valid input and resolve to `Duration::MAX`.
+///
+/// # Arguments
+///
+/// * `src` - A string slice that contains a f64 value
+///
+/// # Errors
+///
+/// This function will return an error when parsing fails (See also [`f64::from_str`]) or the `src`
+/// was negative (-0.0 counts as not negative).
+///
+/// # Examples
+///
+/// ```ignore
+/// use std::time::Duration;
+///
+/// let duration = parse_duration("+1.09e1").unwrap();
+/// assert_eq!(duration, Duration::new(10, 900_000_000));
+/// ```
+///
+/// We accept the same max values like gnu's tail on the command line by parsing into `f64`.
+/// Internally we can only use [`Duration::MAX`] and cap the seconds at [`u64::MAX`]. By doing the
+/// conversion like below, we avoid the panic when using [`Duration::from_secs_f64`] because the
+/// value is too big. The [`Duration::try_from_secs_f64`] isn't yet stable and would also lack the
+/// ability to use the max value in case of an error.
+///
+/// [`f64::from_str`]: https://doc.rust-lang.org/std/primitive.f64.html#method.from_str
+fn parse_duration(src: &str) -> Result<Duration, String> {
+    let trimmed = src.trim();
+    match src.parse::<f64>() {
+        // We're working here with the string representation instead of num.fract() to avoid further
+        // floating point precision problems where we can.
```

Comment:
Converting to a string is lossy too. Are you sure this is worth it? We could also try to copy the unstable implementation in std.

Reply:
> Converting to a string is lossy too.

It's the multiplication with `1_000_000_000` which causes further imprecision. Some of the tests fail with the multiplication. No sure which one it was, but I think something like `1.999999999` resulted in `Duration::new(1, 999_999_998)`. I'm trying to be as precise as possible to make the parsing more predictable.

> Are you sure this is worth it?

Dunno. Sure it's slower than the the multiplication but parsing the sleep interval is a one time action and the string length of the f64 parsed `src` is also managable.

Reply:
I'd imagine that conversion to a string would lead to inaccuracies in different places, but I don't have evidence for that. Let's keep this, but add a link to the tracking issue for `try_from_secs_f64`, so we can periodically check whether it has been stabilized.

Reply:
Nevermind, it's been stabilized in 1.66! Let's put a `Replace with Duration::try_from_secs_f64 once we hit MSRV 1.66` here in a comment.

Reply:
ok, I've added a comment.